sdk/search/search/review/search.api.md

```diff
@@ -246,11 +817,113 @@ export type SuggestResult<T> = {
     readonly text: string;
 } & T;
 
+// @public
+export interface SynonymTokenFilter {
+    expand?: boolean;
+    ignoreCase?: boolean;
+    name: string;
+    odatatype: "#Microsoft.Azure.Search.SynonymTokenFilter";
+    synonyms: string[];
+}
+
+// @public
+export interface TagScoringFunction {
+    boost: number;
+    fieldName: string;
+    interpolation?: ScoringFunctionInterpolation;
+    parameters: TagScoringParameters;
+    type: "tag";
+}
+
+// @public
+export interface TagScoringParameters {
+    tagsParameter: string;
+}
+
+// @public
+export interface TextWeights {
+    weights: {
+        [propertyName: string]: number;
+    };
+}
+
+// @public
+export type TokenCharacterKind = 'letter' | 'digit' | 'whitespace' | 'punctuation' | 'symbol';
+
+// @public
+export interface TokenFilter {
+    name: string;
+    odatatype: "TokenFilter";
+}
+
+// @public
+export type TokenFilterName = 'arabic_normalization' | 'apostrophe' | 'asciifolding' | 'cjk_bigram' | 'cjk_width' | 'classic' | 'common_grams' | 'edgeNGram_v2' | 'elision' | 'german_normalization' | 'hindi_normalization' | 'indic_normalization' | 'keyword_repeat' | 'kstem' | 'length' | 'limit' | 'lowercase' | 'nGram_v2' | 'persian_normalization' | 'phonetic' | 'porter_stem' | 'reverse' | 'scandinavian_normalization' | 'scandinavian_folding' | 'shingle' | 'snowball' | 'sorani_normalization' | 'stemmer' | 'stopwords' | 'trim' | 'truncate' | 'unique' | 'uppercase' | 'word_delimiter';
+
+// @public
+export type TokenFilterUnion = TokenFilter | AsciiFoldingTokenFilter | CjkBigramTokenFilter | CommonGramTokenFilter | DictionaryDecompounderTokenFilter | EdgeNGramTokenFilter | EdgeNGramTokenFilterV2 | ElisionTokenFilter | KeepTokenFilter | KeywordMarkerTokenFilter | LengthTokenFilter | LimitTokenFilter | NGramTokenFilter | NGramTokenFilterV2 | PatternCaptureTokenFilter | PatternReplaceTokenFilter | PhoneticTokenFilter | ShingleTokenFilter | SnowballTokenFilter | StemmerTokenFilter | StemmerOverrideTokenFilter | StopwordsTokenFilter | SynonymTokenFilter | TruncateTokenFilter | UniqueTokenFilter | WordDelimiterTokenFilter;
+
+// @public
+export interface TokenInfo {
+    readonly endOffset: number;
```

Comment:
The docs say this is "character offset", which I don't believe is well defined. Is it graphemes according to whatever version of Unicode exists on the service ala TextAnalytics? Or UTF-16 code units?

Reply:
I'll ask folks in the team who are more familiar with this area.

Reply:
It's the number of characters which I've just learned could be called graphmemes (or rather, we call graphmemes characters colloquially) that might be internally represented by one or more code units.

Reply:
@Yahnoosh I think the bigger concern we have is if this offset would work when indexing into JS strings which are UTF-16

Reply:
I don't think these offsets are meant to be interpreted locally by a client. Think of this as debug spew from the server-side lexical analyzer.